### ISSUE TITLE
Feature: AWS Redshift Support

### ISSUE THREAD

**Issue by dargueta** [xnuinside/simple-ddl-parser#50]
Would it be possible to add support for [AWS Redshift](https://aws.amazon.com/redshift)? It's based on Postgres 8.0 but, being a distributed database with columnar storage, has added some syntax to control how rows are distributed and compressed. You can find a BNF-esque grammar [here](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html).

Currently, this doesn't detect the DDL at all:

```
>>> results = DDLParser("""
...     CREATE TABLE foo.bar(
...         asdf INTEGER ENCODE ZSTD NOT NULL,
...         qwerty VARCHAR(255) ENCODE LZO
...     )
...     SORTSTYLE COMPOUND
...    SORTKEY (qwerty)
...     DISTSTYLE EVEN
...     ;
... """).run()
>>> results
[]
```

It'd be great if we could get the information about encoding, sortkey, distkey, etc. with a "redshift" output mode.

**Comment by xnuinside** [comment]
@dargueta, hi. Yes, why not. Let me add the support of Redshift statements. I will start from your example.

**Comment by xnuinside** [comment]
@dargueta I cannot find SORTSTYLE in docs https://docs.amazonaws.cn/en_us/redshift/latest/dg/r_CREATE_TABLE_NEW.html#r_CREATE_TABLE_NEW-synopsis, see only sortkeys

**Comment by xnuinside** [comment]
`
ddl = """

 CREATE TABLE foo.bar(
     asdf INTEGER ENCODE ZSTD NOT NULL,
     qwerty VARCHAR(255) ENCODE LZO
 )
COMPOUND SORTKEY (qwerty)
 DISTSTYLE EVEN
 ;
"""

result = DDLParser(ddl).run(group_by_type=True, output_mode="redshift")`
will produce now: 

{'domains': [],
 'schemas': [],
 'sequences': [],
 'tables': [{'DISTSTYLE': 'EVEN',
             'alter': {},
             'checks': [],
             'columns': [{'check': None,
                          'default': None,
                          'encode': 'ZSTD',
                          'name': 'asdf',
                          'nullable': False,
                          'references': None,
                          'size': None,
                          'type': 'INTEGER',
                          'unique': False},
                         {'check': None,
                          'default': None,
                          'encode': 'LZO',
                          'name': 'qwerty',
                          'nullable': True,
                          'references': None,
                          'size': 255,
                          'type': 'VARCHAR',
                          'unique': False}],
             'distkey': None,
             'diststyle': None,
             'encode': None,
             'index': [],
             'partitioned_by': [],
             'primary_key': [],
             'schema': 'foo',
             'sortkey': {'keys': ['qwerty'], 'type': 'COMPOUND'},
             'table_name': 'bar',
             'tablespace': None,
             'temp': False}],
 'types': []}`
`

released in 0.18.0 https://pypi.org/project/simple-ddl-parser/ , also added output_mode="redshift"

to more details and tests examples please check: https://github.com/xnuinside/simple-ddl-parser/pull/51 

I close the issue, if you will find any bugs or need to add more statements - please open new issue.

**Comment by dargueta** [comment]
Thanks!
